### ISSUE TITLE
ERROR TypeError: Cannot read property 'naturalWidth' of null

### ISSUE THREAD

**Issue by shahmeet999** [Mawi137/ngx-image-cropper#271]
I face this issue while I am loading the cropped image into the image cropper, please give the solution ASAP.

Thank You

![image](https://user-images.githubusercontent.com/20349696/74634825-b4455200-518a-11ea-9d29-0b3fc775a2bb.png)

@Mawi137

**Comment by shahmeet999** [comment]
> I face this issue while I am loading the cropped image into the image cropper, please give the solution ASAP.
> 
> Thank You
> 
> ![image](https://user-images.githubusercontent.com/20349696/74634825-b4455200-518a-11ea-9d29-0b3fc775a2bb.png)

@Mawi137

**Comment by iiceman40** [comment]
I am having the same issue when using the cropper with imageBase64 as input. It only seems to happend when I revisit my view a second time. The first time around everything seems fine.

PS: As far as I can tell everything works even if the error pops up.

**Comment by Mawi137** [comment]
Do you use display: none to hide the cropper?

**Comment by iiceman40** [comment]
In my case I did not hide the cropper with display: none. 

Not sure if this might be important: I am on Angular 9, with ivy enabled.

**Comment by shahmeet999** [comment]
No I am not hide the cropper it throws error when take input as base64 of image.

**Comment by Mawi137** [comment]
@iiceman40 What's on line 481? :)

@shahmeet999 Could you provide a repo or stackblitz to recreate the issue?

**Comment by iiceman40** [comment]
![error](https://user-images.githubusercontent.com/3456379/75495399-bbf0cc00-59be-11ea-9f21-7371162c98cf.JPG)
Not sure if that helps, but that's line 481 ^^

**Comment by Mawi137** [comment]
I just released version `3.1.0` and added a console.error statement there so we can actually see the error :D

**Comment by iiceman40** [comment]
Alright, thanks for the update. I tried the new version and i get this:
![error2](https://user-images.githubusercontent.com/3456379/75607740-8bae4800-5afa-11ea-99e9-6644d49b0546.JPG)

I figured out that I can avoid all errors if I put my *ngIf="originalImageBase64" not on the cropper component but an ng-container around it. I only tired that with the latest version 3.1.0 so no guarantee that it works in the previous version, too (but I guess it would have worked there the same way?).

So for me everything is cool now, thanks a lot @Mawi137  I really appreciate the help! :)

**Comment by wolfbomb** [comment]
@iiceman40, can you share on how you fix this? I am having the same issue as yours. I tried your *ngIf fix but it didn't help me. Am i missing something? I am using the latest version of the cropper which is 3.1.1. Hope you or anyone can advise me with this.

I am using Angular CLI: 8.3.22

![image](https://user-images.githubusercontent.com/4049788/75988935-50ae7900-5f2d-11ea-8ef1-b70c8efb399c.png)

**Comment by snakenstein** [comment]
I have had the same issue right now on v.3.1.1 and Angular 9. 
Hiding cropper (`[hidden]="imageBase64Data"`) instead of removing it from the DOM (`*ngIf="imageBase64Data"`) solved the problem

**Comment by kyleabens** [comment]
None of the above solutions worked for me.  Doesn't matter if I use `hidden` or `*ngIf`.  Anytime I load it for a second time I get 2 errors:

```
Error: Uncaught (in promise): [object Undefined]
    at resolvePromise
```

```
TypeError: Cannot read property 'naturalWidth' of null
    at Image.transformedImage.onload
```

**Comment by iiceman40** [comment]
@wolfbomb sorry, I don't really see the difference to what I did. I'll check again what version I use (I think I downgraded because I was debugging another issue), maybe there is a difference. Maybe one of you guys should create a stackblitz that shows the issue so we can take a look together? :)

**Comment by shahmeet999** [comment]
![image](https://user-images.githubusercontent.com/20349696/76206558-425ca780-6222-11ea-849e-da724cd053a2.png)

I am still getting the above error even if I upgrade the library package.

Here I can attach my code.


` 
<div class="row" *ngIf="selectedImage">
<div class="col-md-12" style="height: 400px; width: 200px;">
        <image-cropper #imgCropper [imageBase64]="selectedImage || ''" [maintainAspectRatio]="false"
          (imageCropped)="imageCropped($event)" (imageLoaded)="imageLoaded()" (cropperReady)="cropperReady()"
          (loadImageFailed)="loadImageFailed()" (cropperReady)="cropperReady($event)"
          [style.display]="showCropper ? null : 'none'" [alignImage]="'center'">
        </image-cropper>
      </div>
</div>
`

**Comment by Mawi137** [comment]
Should be fixed in version `3.1.2`!
